### DISCUSSION TITLE
How to publish multiple parameters (URI) on the same port (multi resolution's outputs)

### DISCUSSION THREAD

**Discussion by whitetiger21022014** [bluenviron/mediamtx#1540] [Questions]
Hello,
Is there any option, any solution that I can publish as below

1 MP4 video --- > rtsp-simple-server ---> NVR

Publish main (e.g 1920x1080) and sub stream (e.g 352x288) on the same port, multi resolution's outputs

Main stream: rtsp://192.168.1.112:8554/cam/realmonitor?channel=1&subtype=0
Sub stream:  rtsp://192.168.1.112:8554/cam/realmonitor?channel=1&subtype=1

**ffmpeg example:**
ffmpeg -stream_loop -1 -i test-demo.mp4 -filter_complex \
"[0:v]split=2[s0][s1];[s0]scale=1280:-2[v0];[s1]scale=960:-2[v1]" \
-map "[v0]" -map "[v1]" -c:v hevc -flags +global_header -f tee \
"[select=\'v:0,a\':f=rtsp:onfail=ignore]rtsp://localhost:8554/cam/realmonitor?channel=1&subtype=0|[select=\'v:1,a\':f=rtsp:onfail=ignore]rtsp://localhost:8554/cam/realmonitor?channel=1&subtype=1"

Thanks,

**Comment by aler9** [comment]
Hello, you have to use different paths:

```
ffmpeg -stream_loop -1 -i test-demo.mp4 -filter_complex
"[0:v]split=2[s0][s1];[s0]scale=1280:-2[v0];[s1]scale=960:-2[v1]"
-map "[v0]" -map "[v1]" -c:v hevc -flags +global_header -f tee
"[select='v:0,a':f=rtsp:onfail=ignore]rtsp://localhost:8554/cam/realmonitor_ch1_subtype0|
[select='v:1,a':f=rtsp:onfail=ignore]rtsp://localhost:8554/cam/realmonitor_ch1_subtype1"
```

In the server, streams are splitted by path. Query parameters (?channel=1&subtype=0) can't be used to split paths since there are some RTSP clients that automatically attach them to every URL, so they would be unable to publish or read correctly.

[a comment by an automated account is not shown]
